erpnext/patches/v13_0/set_values_in_pick_list.py

```diff
@@ -0,0 +1,15 @@
+import frappe
+def execute():
+	frappe.reload_doctype("Pick List")
+	pick_lists= frappe.get_all("Pick List",filters= {
+		"purpose": "Delivery"
+	}, fields= ["name","customer"])
+	for pick_list in pick_lists:
+		order_delivery_date = []
+		sales_order = frappe.get_value("Pick List Item",{"parent": pick_list.name}, "sales_order")
```

Comment:
What if there are multiple Pick Lists for a sales order?

Reply:
it will assign 'delivery date' and 'customer name' to all the pick list. As it is picking all the pick list 1 by 1.

Reply:
It's still not reliable though. Passing filters to `get_value` means that it'll find the first match and get the value from that document, and it's possible the `sales_order` field is empty

Reply:
for that i am checking that if sales order field is empty it will skip that pick list

Reply:
That's what I mean, it could be empty for just one item in the Pick List, whereas it might be set for other items. You should instead get a list of unique sales orders from the items:

```suggestion
		sales_order = frappe.get_all("Pick List Item", filters={"parent": pick_list.name}, fields=["sales_order"], distinct=True)
```